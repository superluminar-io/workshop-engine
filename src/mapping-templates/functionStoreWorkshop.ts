export const request = (tableName: string): string => `
  #set($ctx.stash = {})
  $util.qr($ctx.stash.put("workshopId", $util.autoId()))
  $util.qr($ctx.stash.put("createdAt", $util.time.nowISO8601()))

  #set($entities = [])
  $util.qr($entities.add($util.dynamodb.toMap({ "workshopId": "$ctx.stash.workshopId", "node": "workshop", "title": "$ctx.arguments.input.title", "awsAccountId": "$ctx.arguments.input.awsAccountId", "createdAt": "$ctx.stash.createdAt" }).M))
  #foreach($attendee in $ctx.arguments.input.attendees)
    $util.qr($entities.add($util.dynamodb.toMap({ "workshopId": "$ctx.stash.workshopId", "node": "attendee#$attendee", "awsAccountId": "$ctx.arguments.input.awsAccountId", "createdAt": "$ctx.stash.createdAt" }).M))
  #end
  {
    "version" : "2018-05-29",
    "operation" : "BatchPutItem",
    "tables" : {
      "${tableName}": $utils.toJson($entities)
    }
  }
`;

export const response = `
  {
    "id": "$ctx.stash.workshopId",
    "title": "$ctx.arguments.input.title",
    "createdAt": "$ctx.stash.createdAt",
    "awsAccountId": "$ctx.arguments.input.awsAccountId",
  }
`;
