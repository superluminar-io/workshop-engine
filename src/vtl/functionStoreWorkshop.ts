export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "PutItem",
    "key": {
      "id": $util.dynamodb.toDynamoDBJson($util.autoId())
    },
    "attributeValues" : $util.dynamodb.toMapValuesJson({
      "node": "workshop",
      "createdAt": "$util.time.nowISO8601()",
      "title": "$ctx.arguments.input.title",
      "description": "$util.defaultIfNullOrEmpty($ctx.arguments.input.description, '')",
      "awsAccountId": "$util.defaultIfNullOrEmpty($ctx.arguments.input.awsAccountId, '')",
      "attendees": $ctx.arguments.input.attendees
    })
  }
`;

export const response = `
  $util.toJson($ctx.result)
`;
