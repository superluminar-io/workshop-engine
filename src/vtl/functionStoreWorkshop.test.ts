import Parser from 'appsync-template-tester';
import { request, response } from './functionStoreWorkshop';

jest.mock('uuid', () => ({ v4: () => 'beb79a10-437e-48df-96e2-9cd87fdc063f' }));
jest.mock('moment-timezone', () => ({ utc: () => new Date('2022-04-22T19:38:10.703Z') }));

describe('request mapping', () => {
  describe('input w/ aws account id', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const input = {
        title: 'Serverless Beginner Workshop',
        awsAccountId: '298097285261',
        attendees: [
          'me@xample.com',
          'you@xample.com',
        ],
      };

      const context = {
        arguments: {
          input,
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('input w/o aws account id', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const input = {
        title: 'Serverless Beginner Workshop',
        attendees: [
          'me@xample.com',
          'you@xample.com',
        ],
      };

      const context = {
        arguments: {
          input,
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(response);

    const context = {
      result: {
        any: 'data',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
