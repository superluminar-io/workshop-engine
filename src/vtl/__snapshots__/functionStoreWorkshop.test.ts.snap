// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping input w/ aws account id should match snapshot 1`] = `
Object {
  "attributeValues": Object {
    "attendees": Object {
      "L": Array [
        Object {
          "S": "me@xample.com",
        },
        Object {
          "S": "you@xample.com",
        },
      ],
    },
    "awsAccountId": Object {
      "S": "298097285261",
    },
    "createdAt": Object {
      "S": "2022-04-22T19:38:10.703Z",
    },
    "node": Object {
      "S": "workshop",
    },
    "title": Object {
      "S": "Serverless Beginner Workshop",
    },
  },
  "key": Object {
    "id": Object {
      "S": "beb79a10-437e-48df-96e2-9cd87fdc063f",
    },
  },
  "operation": "PutItem",
  "version": "2017-02-28",
}
`;

exports[`request mapping input w/o aws account id should match snapshot 1`] = `
Object {
  "attributeValues": Object {
    "attendees": Object {
      "L": Array [
        Object {
          "S": "me@xample.com",
        },
        Object {
          "S": "you@xample.com",
        },
      ],
    },
    "awsAccountId": Object {
      "S": "",
    },
    "createdAt": Object {
      "S": "2022-04-22T19:38:10.703Z",
    },
    "node": Object {
      "S": "workshop",
    },
    "title": Object {
      "S": "Serverless Beginner Workshop",
    },
  },
  "key": Object {
    "id": Object {
      "S": "beb79a10-437e-48df-96e2-9cd87fdc063f",
    },
  },
  "operation": "PutItem",
  "version": "2017-02-28",
}
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
